### PR TITLE
Tremolo stem direction fix

### PR DESCRIPTION
**Pull request by asattely**
Problem:
tremolo-before

Notes with tremolos with no beams should be made to keep consistent direction, similar to behavior from beams:
tremolo-after

In the future (likely post-4.0), these tremolos will be moved in stem direction to be more consistent with where beams would normally be. However, we don't have enough time to do that at the moment as it will take a lot of engineering to factor out that behavior from the `Beam` class and be made to work for `Tremolo` as well, perhaps with a helper class of some kind.

### CODE REVIEW

File: src/importexport/musicxml/tests/data/testTremolo.xml [hunk 1]
```diff
@@ -293,7 +293,7 @@
           <actual-notes>2</actual-notes>
           <normal-notes>1</normal-notes>
           </time-modification>
-        <stem>down</stem>
+        <stem>up</stem>
```
[the remark this answers is not in the document]
Comment (RomanPudashkin) [reply src/importexport/musicxml/tests/data/testTremolo.xml:296]: It would be great if you could add new tests that would guard your fix and protect us from regressions

[the remark this answers is not in the document]
Comment (asattely) [reply src/importexport/musicxml/tests/data/testTremolo.xml]: Apologies for the delay, I have been away for a few days.

This utest change was a dumb mistake by me; after checking `testTwoNoteTremoloTuplet.xml`, which contained what now amounts to an illegal two note tremolo (a tremolo consisting of two notes with opposite stem directions) I thought I had checked both failing tests and that they both had the same problem. They did not, so all changes to `testTremolo.xml` have been reverted.

There is still a change to `testTwoNoteTremoloTuplet.xml`, because it contains this:
image
...which is not a valid tremolo anymore. And I might also add that if this xml tests tremolo tuplet and fails on tremolos that are not tuplets, it's a very strangely designed test.

I will add a utest that deals with flipping one note of a two-note tremolo ensuring that the other is also flipped. This could take me some time as I'm not 100% familiar with testing of commands.

[the remark this answers is not in the document]
Comment (asattely) [reply src/importexport/musicxml/tests/data/testTremolo.xml]: I guess now thinking about it, the test itself might have to be modified anyway to A) make more sense and test the things it's supposed to test and nothing more, or at least B) have a reference file to show the differences between the musicxml and the musescore treatment, which will be different from the imported data (with explicit stem directions). What do you think?

File: src/engraving/utests/beam_tests.cpp [hunk 2]
```diff
@@ -225,3 +226,34 @@ TEST_F(BeamTests, flipBeamStemDir)
...
+{
+    MasterScore* score = ScoreRW::readScore(BEAM_DATA_DIR + "flipTremoloStemDir.mscx");
+    EXPECT_TRUE(score);
+
+    Measure* m1 = score->firstMeasure();
+    ChordRest* cr = toChordRest(m1->findSegment(SegmentType::ChordRest, m1->tick())->element(0));
+    Tremolo* t = toChord(cr)->tremolo();
+    Chord* c1 = t->chord1();
+    Chord* c2 = t->chord2();
+    t->chord1()->up();
```
Comment (HemantAntony) [inline src/engraving/utests/beam_tests.cpp:247]: Why is this here? This just return a bool whether its up or down. Seems pretty pointless. Did you forget to do something here? Maybe this could be the source of your problems?

Comment (asattely) [reply src/engraving/utests/beam_tests.cpp:247]: Good catch, but that isn't the source of the problem. The problem is that `t->up()` and `c1->up()` and `c2->up()` are all true, but after selecting the first chord and pressing x (executing `cmdFlip()` in other words) will flip `c1`, `c2`, and `t`. The line you identified can be safely removed, and I shall do that in the next push.
